can/logger.py

```diff
@@ -213,13 +213,13 @@ def main() -> None:
     print(f"Connected to {bus.__class__.__name__}: {bus.channel_info}")
     print(f"Can Logger (Started on {datetime.now()})")
 
-    options = {"append": results.append}
     if results.file_size:
+        options = {"append": results.append}
         logger = SizedRotatingLogger(
             base_filename=results.log_file, max_bytes=results.file_size, **options
         )
     else:
-        logger = Logger(filename=results.log_file, **options)  # type: ignore
+        logger = Logger(filename=results.log_file)  # type: ignore
```

Comment:
This change won't be necessary once https://github.com/hardbyte/python-can/pull/1361 is merged.